mempool/mempool.go

```diff
@@ -1307,9 +1452,28 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
 		return nil, err
 	}
 
+	if isTicket && mp.hasMempoolInput(tx) {
+		mp.stageTransaction(tx)
+		log.Debugf("Accepted ticket %v with mempool dependency "+
+			"into stage pool", txHash)
+		return nil, nil
+	}
+
 	// Add to transaction pool.
 	mp.addTransaction(utxoView, tx, txType, bestHeight, txFee)
 
+	if !isNew && txType == stake.TxTypeRegular {
```

Comment:
Add a comment to explain that regular transactions re-entering the pool bump any unmined tickets from the main txpool back to the staged pool.